### PR TITLE
Create movie list

### PR DESCRIPTION
**Pull request by wroblewskisoftware**
I downloaded an object containing a list of popular movies and an object containing a list of genres from the API. I rendered the video tiles and made the page layout. The effect can be seen by inserting it into the App component.

```
import { MovieList } from "./features/MovieList";

const App = () => <MovieList />;

export default App;
```

### CODE REVIEW

File: src/common/MovieTile/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          </Genres>
+        )}
+        <VotesContainer>
+          <StyledStarIcon />
+          <Mark>{vote_average}</Mark>
+          <VotesNumber>{vote_count} votes</VotesNumber>
+        </VotesContainer>
+      </Decsription>
+    </StyledMovieTile>
+  );
```
Comment (sofyahreksoftware) [inline src/common/MovieTile/index.js]: I think it's better to keep this component separate from the MovieList logic, as its structure might be useful for displaying films from other APIs in the future. The main logic can be transferred to its props in MovieList component.
```js
import { nanoid } from "@reduxjs/toolkit";
import { ReactComponent as PlaceholderIcon } from "../../assets/placeholderMovieIcon.svg";

import { ImagePlaceholder } from "../ImagePlaceholder/styled";
import {
  StyledMovieTile,
  Image,
  Decsription,
  Title,
  Year,
  Genres,
  Genre,
  VotesContainer,
  StyledStarIcon,
  VotesNumber,
  Mark,
} from "./styled";

export const MovieTile = ({
  poster,
  title,
  year,
  genres,
  mark,
  votesNumber,
}) => {
  return (
    <StyledMovieTile>
      {poster ? (
        <Image src={poster} alt="Movie Poster" />
      ) : (
        <ImagePlaceholder>
          <PlaceholderIcon />
        </ImagePlaceholder>
      )}
      <Decsription>
        {title && <Title>{title}</Title>}
        {year && <Year>{year}</Year>}
        {genres && (
          <Genres>
            {genres.map((genre) => (
              <Genre key={nanoid()}>{genre?.name}</Genre>
            ))}
          </Genres>
        )}
        <VotesContainer>
          <StyledStarIcon />
          <Mark>{mark}</Mark>
          <VotesNumber>{votesNumber} votes</VotesNumber>
        </VotesContainer>
      </Decsription>
    </StyledMovieTile>
  );
};

File: src/features/MovieList/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <StyledMain>
+      <StyledHeader>Popular movies</StyledHeader>
+      <StyledList>
+        {moviesList?.map((movie) => (
+          <StyledItem key={movie.id}>
+            <MovieTile {...movie} />
+          </StyledItem>
+        ))}
```
Comment (sofyahreksoftware) [inline src/features/MovieList/index.js]: The main logic of MovieTile with props in MovieList component according to moving main logics from MovieTile component:
```js
  {movies.map((movie) =>
        movie.results.map((result) => (
          <MovieTile
            key={result.id}
            poster={`https://image.tmdb.org/t/p/w500${result.poster_path}`}
            title={result.title}
            year={result.release_date.split("-")[0]}
            genres={result.genre_ids.map((id) =>
              genres.find((genre) => genre.id === id)
            )}
            mark={result.vote_average.toFixed(1)}
            votesNumber={result.vote_count}
          />
        ))
      )}

File: src/features/MovieList/moviesSlice.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const moviesSlice = createSlice({
+  name: "movies",
+  initialState: {
+    movies: [],
+    genres: [],
+    isLoading: false,
+    isError: false,
```
Comment (sofyahreksoftware) [inline src/features/MovieList/moviesSlice.js]: I think it's better to change **isLoading** and **isError** variables to **fetchMoviesStatus** with initial state undefined and reduce the number of selectors from two to 1:)

File: src/features/MovieList/moviesSlice.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+export const selectMoviesState = (state) => state.movies;
+export const selectMoviesObject = (state) => selectMoviesState(state).movies;
+export const selectLoadingStatus = (state) =>
+  selectMoviesState(state).isLoading;
+export const selectErrorStatus = (state) => selectMoviesState(state).isError;
+
+export const selectMoviesList = (state) =>
+  selectMoviesObject(state).movieList?.results;
+export const selectMoviesGenres = (state) =>
+  selectMoviesObject(state).genreList.genres;
```
Comment (sofyahreksoftware) [inline src/features/MovieList/moviesSlice.js]: I think you can use map method on movies returned by **selectMoviesObiect** to reduce an amount of selectors.

**Review by sofyahreksoftware — CHANGES_REQUESTED** [review]
Good job! Just a few thing🤗

File: src/features/MovieList/moviesSlice.js (hunk 3 above)
Comment (wroblewskisoftware) [reply src/features/MovieList/moviesSlice.js]: Thank you, there is actually a simpler way to do it.

File: src/features/MovieList/index.js (hunk 2 above)
Comment (wroblewskisoftware) [reply src/features/MovieList/index.js]: It's actually better to move the logic outside the MovieTile component, thank you for the tip.

File: src/features/MovieList/moviesSlice.js (hunk 4 above)
Comment (wroblewskisoftware) [reply src/features/MovieList/moviesSlice.js]: I admit that I don't understand what you mean.

File: src/features/MovieList/moviesSlice.js [hunk 5]
```diff
@@ -5,21 +5,17 @@ const moviesSlice = createSlice({
   initialState: {
     movies: [],
     genres: [],
-    isLoading: false,
-    isError: false,
+    loadingStatus: "loading",
```
Comment (sofyahreksoftware) [inline src/features/MovieList/moviesSlice.js]: I think it's better to name this variable as fetchingStatus/fetchMoviesStatus:)

File: src/features/MovieList/index.js [hunk 6]
```diff
@@ -13,32 +13,39 @@ import { StyledMain, StyledHeader, StyledList, StyledItem } from "./styled";
...
+                    genreList.find((genre) => genre.id === id)
+                  )}
+                  mark={movie.vote_average.toFixed(1)}
+                  votes={movie.vote_count}
+                />
+              </StyledItem>
+            ))}
+          </StyledList>
+        </>
+      )}
```
Comment (sofyahreksoftware) [inline src/features/MovieList/index.js]: Now it looks great!🥰🥰

**Review by sofyahreksoftware — APPROVED** [review]
Good job. Now it looks better🤗The only note is changing the name of **loadingStatus** to **fetchMoviesStatus**/**fetchDataStatus** for better code understanding!

**Pull request merged**